### DISCUSSION TITLE
API for faces

### DISCUSSION THREAD

**Discussion by JeCheeseSmith** [nextcloud/recognize#884] [Q&A] [answered]
Hi! I was wondering if there's a way to extract the face data with names or photo references or so? I would like to use the people found through my facerecognition to try build a generic tree of my own family!

**Comment by marcelklehr** [comment, marked as answer]
Hey @JeCheeseSmith :wave: 

You can use the WebDAV API. It is sadly not documented yet, but here's a quick overview

 * `remote.php/dav/recognize/<username>/faces/`
    * contains the people folders, named with the name you gave them in the UI
 *  `remote.php/dav/recognize/<username>/faces/<facename>/`
   * contains the photos for this person
   * has the following additional properties:
     * nc:nbItems - the number of photos
     * nc:face-preview-image - a json struct with information about the preview image for this person
  *  `remote.php/dav/recognize/<username>/faces/<facename>/<xxx-filename>.jpg`
    * the photo
    * has the following aditional properties:
      * nc:face-detections - a json struct containing information about the faces detected in this photo

**Reply by JeCheeseSmith** [reply]
Hi! Thanks for the information :) I'm kind of a newbie into Nextcloud but I will try what i can make of it!

**Comment by phil-lipp** [comment]
> Hi! Thanks for the information :) I'm kind of a newbie into Nextcloud but I will try what i can make of it!

If you're accustomed to Python you can use [this API wrapper](https://pypi.org/project/nextcloud-api-wrapper/) to query the API. There is also a really helpful [example script](https://github.com/luffah/nextcloud-API/blob/main/examples/files.py).
